phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexScrutinyToolIT.java

```diff
@@ -81,681 +86,844 @@
+                String indexNameTenant = generateUniqueName();
+                String indexNameGlobal = generateUniqueName();
+                String viewIndexTableName = "_IDX_" + MULTI_TENANT_DATA_TABLE_NAME;
+
+                props.setProperty(PhoenixRuntime.TENANT_ID_ATTRIB, tenantId);
+                connTenant = DriverManager.getConnection(getUrl(), props);
+                String
+                        createTblStr =
+                        "CREATE TABLE %s (COL1 VARCHAR(15) NOT NULL,ID INTEGER NOT NULL" + ", NAME VARCHAR, CONSTRAINT PK_1 PRIMARY KEY (COL1, ID)) MULTI_TENANT=true";
+                String createViewStr = "CREATE VIEW %s AS SELECT * FROM %s";
+
+                String upsertQueryStr = "UPSERT INTO %s (COL1, ID, NAME) VALUES('%s' , %d, '%s')";
+                String createIndexStr = "CREATE INDEX %s ON %s (NAME) ";
+
+                createTestTable(getUrl(), String.format(createTblStr, MULTI_TENANT_DATA_TABLE_NAME));
+
+                connTenant.createStatement().execute(
+                        String.format(createViewStr, tenantViewName, MULTI_TENANT_DATA_TABLE_NAME));
+
+                String idxStmtTenant = String.format(createIndexStr, indexNameTenant, tenantViewName);
+                connTenant.createStatement().execute(idxStmtTenant);
+
+                connTenant.createStatement()
+                        .execute(String.format(upsertQueryStr, tenantViewName, tenantId, 1, "x"));
+                connTenant.commit();
+
+                String[] argValues =
+                        getArgValues("", tenantViewName, indexNameTenant, 10L,
+                                SourceTable.INDEX_TABLE_SOURCE, false, null, null, tenantId,
+                                EnvironmentEdgeManager.currentTimeMillis());
+
+                List<Job> completedJobs = runScrutiny(argValues);
+                // Sunny case, both index and view are equal. 1 row
+                assertEquals(1, completedJobs.size());
+                for (Job job : completedJobs) {
+                    assertTrue(job.isSuccessful());
+                    Counters counters = job.getCounters();
+                    assertEquals(1, getCounterValue(counters, VALID_ROW_COUNT));
+                    assertEquals(0, getCounterValue(counters, INVALID_ROW_COUNT));
+                }
 
-    private List<Job> runScrutiny(String[] cmdArgs) throws Exception {
-        IndexScrutinyTool scrutiny = new IndexScrutinyTool();
-        Configuration conf = new Configuration(getUtility().getConfiguration());
-        scrutiny.setConf(conf);
-        int status = scrutiny.run(cmdArgs);
-        assertEquals(0, status);
-        for (Job job : scrutiny.getJobs()) {
-            assertTrue(job.waitForCompletion(true));
-        }
-        return scrutiny.getJobs();
-    }
+                ///////
+                // Global view on multi-tenant table should work too
+                connGlobal.createStatement().execute(
```

Comment:
@gokceni  Would be good to divide this big testTenantId up into several smaller tests, since it's going through several distinct use cases of "create / populate table", "run scrutiny", then "verify scrutiny". This would make it more similar to the other non-tenant test suite and give developers more clarity about what's failing if a test fails.

Other than that, I think this is just about ready to go.